tdrs-backend/tdpservice/parsers/row_schema.py

```diff
@@ -188,12 +189,14 @@ class SchemaManager:
 
     def __init__(self, schemas):
         self.schemas = schemas
+        self.datafile = None
 
     def parse_and_validate(self, line, generate_error):
         """Run `parse_and_validate` for each schema provided and bubble up errors."""
         records = []
 
         for schema in self.schemas:
+            schema.datafile = self.datafile
```

Comment:
not sure i like setting the datafile like this as a side-effect of parsing+validation executing. is there a way to move this to the `__init__`?